### PR TITLE
Camera auto connect on startup / automatically reconnect

### PR DESCRIPTION
**Pull request by stefan-sherwood**

### CODE REVIEW

**Comment by stefan-sherwood** [comment]
This is of interest to @rhysmorgan134

File: src/app/tabs/camera.cpp [hunk 1]
```diff
@@ -22,9 +25,20 @@ CameraTab::CameraTab(QWidget *parent) : QWidget(parent)
     layout->addWidget(this->local_camera_widget());
     layout->addWidget(this->network_camera_widget());
 
-    connect(this, &CameraTab::disconnected, [layout]() { layout->setCurrentIndex(0); });
+    connect(this, &CameraTab::disconnected, [layout,this]() {
```
Comment (rsjudka) [inline src/app/tabs/camera.cpp:28]: i dont see this signal ever being emitted

File: src/app/config.cpp [hunk 2]
```diff
@@ -34,6 +34,8 @@ Config::Config()
     this->cam_local_device = this->ia_config.value("Camera/local_device").toString();
     this->cam_is_network = this->ia_config.value("Camera/is_network").toBool();
     this->cam_local_format_override = this->ia_config.value("Camera/local_format_override").value<QVideoFrame::PixelFormat>();
+    this->cam_autoconnect = this->ia_config.value("Camera/automatically_reconnect").toBool();
+    this->cam_autoconnect_time_secs = this->ia_config.value("Camera/auto_reconnect_time_secs", 6).toInt();
```
Comment (rsjudka) [inline src/app/config.cpp:38]: do we even want this to be configurable? I dont see where the user can set it

Comment (stefan-sherwood) [reply src/app/config.cpp:38]: It's not in the user interface. I'm not sure what a good value is and whether or not the right answer will vary from machine to machine so I want people to be able to change it (via the .ini file) if this value is bothersome.

File: src/app/tabs/camera.cpp (hunk 1 above)
Comment (stefan-sherwood) [reply src/app/tabs/camera.cpp:28]: Line 313 of this file

Comment (rsjudka) [reply src/app/tabs/camera.cpp:28]: ah it was hidden by the diff lol

File: src/app/tabs/camera.cpp [hunk 3]
```diff
@@ -22,9 +25,20 @@ CameraTab::CameraTab(QWidget *parent) : QWidget(parent)
...
+        layout->setCurrentIndex(0);
+        if (this->config->get_cam_autoconnect()) {
```
Comment (rsjudka) [inline src/app/tabs/camera.cpp:30]: what are the changes it just continually tries reconnecting forever and ever?

like if the reconnection happens, and the camera just isnt available, will `UnloadedStatus` be set, thus causing the reconnection to happen again?

Comment (stefan-sherwood) [reply src/app/tabs/camera.cpp:30]: Yes, it tries reconnecting every X seconds (6 by default) forever, or until you uncheck the "automatically connect" checkbox.

The most important feature here is the connect on startup. The reconnect will probably only happen if someone unplugs a camera (like USB) or there's a network glitch.

**Pull request merged**